test/integ/quote-to-ratio.test.ts

```diff
@@ -0,0 +1,357 @@
+import DEFAULT_TOKEN_LIST from '@uniswap/default-token-list';
+import { Currency, Ether, Fraction } from '@uniswap/sdk-core';
+import {
+  CachingTokenListProvider,
+  NodeJSCache,
+} from '@uniswap/smart-order-router';
+import axios, { AxiosResponse } from 'axios';
+import { parseUnits } from 'ethers/lib/utils';
+import JSBI from 'jsbi';
+import NodeCache from 'node-cache';
+import qs from 'qs';
+import {
+  QuoteToRatioQueryParams,
+  QuoteToRatioResponse,
+  ResponseFraction,
+} from '../../lib/handlers/quote-to-ratio/schema/quote-to-ratio-schema';
+import { FeeAmount, getMinTick, getMaxTick, TICK_SPACINGS } from '../utils/ticks';
+import { absoluteValue } from '../utils/absoluteValue';
+
+const tokenListProvider = new CachingTokenListProvider(
+  1,
+  DEFAULT_TOKEN_LIST,
+  new NodeJSCache(new NodeCache())
+);
+
+const API = `${process.env.UNISWAP_ROUTING_API!}quoteToRatio`;
+
+// Try to parse a user entered amount for a given token
+async function parseAmount(
+  value: number,
+  tokenAddress: string
+): Promise<string> {
+  const decimals = (await tokenStringToCurrency(tokenAddress))?.decimals;
+  return parseUnits(value.toString(), decimals).toString();
+}
+
+async function tokenStringToCurrency(tokenString: string): Promise<Currency | undefined> {
+  const isAddress = (s: string) => s.length == 42 && s.startsWith('0x');
+
+  let token: Currency | undefined
+  if (
+    tokenString == 'ETH' ||
+    tokenString.toLowerCase() == '0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee'
+  ) {
+    token = Ether.onChain(1);
+  } else if (isAddress(tokenString)) {
+    token = await tokenListProvider.getTokenByAddress(tokenString);
+  }
+
+  if (!token) {
+    token = await tokenListProvider.getTokenBySymbol(tokenString);
+  }
+
+  if (!token) {
+    throw new Error('could not find token')
+  }
+
+  return token;
+
+}
+
+function parseFraction(fraction: ResponseFraction): Fraction {
+  return new Fraction(JSBI.BigInt(fraction.numerator), JSBI.BigInt(fraction.denominator))
+}
+
+describe('quote-to-ratio', () => {
```

Comment:
Can we get some 4xx failure case tests?

Good to test some of the basic validation cases just to check that validation is hooked up (e.g. incorrect address returns 4xx etc.), as well as cases where we don't find a swap? (maybe we can set maxIterations to 0 or 1? or maybe passing in token0Balance = 0 and token1Balance = 0?)

Also what happens if I want to do a range order, but I already have 100% of the token required and 0 of the token not required? Do we still recommend a swap? Do we return 200 or 4xx in that case?